clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3151,11 +3151,21 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
       return;
     ZDimExpr = ZDim.get();
 
+    // If the declaration has an [[intel::num_simd_work_items()]] attribute,
+    // check to see if the first argument (OpenCL spelling) or last argument
+    // (SYCL spelling) can be evenly divided by the num_simd_work_items
+    // attribute since first and last argument are only swapped for the
+    // Intel SYCL attributes and not the OpenCL ones.
     if (const auto *A = D->getAttr<SYCLIntelNumSimdWorkItemsAttr>()) {
       int64_t NumSimdWorkItems =
           A->getValue()->getIntegerConstantExpr(Ctx)->getSExtValue();
 
-      if (XDimVal.getZExtValue() % NumSimdWorkItems != 0) {
+      unsigned checkWorkGroupSize =
+          AL.getSyntax() == AttributeCommonInfo::AS_GNU
```

Comment:
We should document clearly that the GNU spelling of this attribute maps to the OpenCL semantics (because otherwise it's extremely unclear as to which semantics you get since there's no vendor information).

Reply:
Also, this code misses the `[[cl::reqd_work_group_size]]` spelling. We should be sure to add tests for both the GNU and CXX spellings so we don't accidentally regress this sort of thing in the future.

Reply:
>>We should be sure to add tests for both the GNU and CXX spellings so we don't accidentally regress this sort of thing in the future.

@AaronBallman , just want to check with you that how can we add tests for GNU spelling here? Am i missing anything?

we can not use [[intel::num_simd_work_items]] and __attribute__((reqd_work_group_size))  together. how can we verify that?

We have  __attribute__((reqd_work_group_size))  and  __attribute__((num_simd_work_items)) in downstream as OpenCL spelling

Reply:
I'm sorry, I'm not certain I understand the questions. You're already testing for the GNU spelling? Did you mean the `[[cl:reqd_work_group_size]]` spelling? You can do something like `(AL.getSyntax() == AttributeCommonInfo::AS_CXX11 || AL.getSyntax() == AttributeCommonInfo::AS_C2x) && AL.hasScope() && AL.getScopeName()->isStr("cl")` to get you there.

Reply:
Sorry i meant to say new test cases of __attribute__((reqd_work_group_size))

Reply:
Sorry @AaronBallman, i was not clear to my question. 
I confused myself with the  syntax of reqd_work_group_size attributes
`kernel __attribute__((reqd_work_group_size(1,1,2))) void kernel13(){}`

>>am I misunderstanding still?

No, you understood correctly. You answered my questions. Thanks.

Reply:
>>We should be sure to add tests for both the GNU and CXX spellings so we don't accidentally regress this sort of thing in the future.

Added test cases, Done.